rust/rope/src/engine.rs

```diff
@@ -222,6 +234,18 @@ impl Engine {
         self.rev_id_counter += 1;
     }
 
+    pub fn compare(&self, base_rev: usize, other_rev: usize) -> bool {
+        let base_subset = self.find_rev(base_rev).map(|rev_index| self.get_subset_from_index(rev_index));
+        let other_subset = self.find_rev(other_rev).map(|rev_index| self.get_subset_from_index(rev_index));
+
+        if let Some(base_subset) = base_subset {
```

Comment:
Sweet. You could write this as `base_subset.is_some() && base_subset == other_subset`.